### PR TITLE
Make cummax consistent with cummin

### PR DESCRIPTION
**Pull request by chipkent**
Make `cummax` consistent with `cummin`.  `cummax` did not handle nulls properly.

### CODE REVIEW

File: engine/function/src/templates/Numeric.ftl [hunk 1]
```diff
@@ -1579,7 +1579,13 @@ public class Numeric {
         result[0] = values[0];
 
         for (int i = 1; i < values.length; i++) {
-            result[i] = compare(result[i-1], values[i]) > 0 ? result[i-1] : values[i];
+            if (isNull(result[i - 1])) {
+                result[i] = values[i];
+            } else if (isNull(values[i])) {
+                result[i] = result[i - 1];
+            } else {
+                result[i] = (${pt.primitive})Math.max(result[i - 1],  values[i]);
```
Comment (kosak) [inline engine/function/src/templates/Numeric.ftl:1587]: Can we remove the extra space before values[i] at the end

File: engine/function/src/templates/Numeric.ftl [hunk 2]
```diff
@@ -1606,10 +1612,18 @@ public class Numeric {
...
             while (vi.hasNext()) {
                 final ${pt.primitive} v = vi.${pt.iteratorNext}();
-                result[i] = compare(result[i-1], v) > 0 ? result[i-1] : v;
+
+                if (isNull(result[i - 1])) {
+                    result[i] = v;
+                } else if (isNull(v)) {
+                    result[i] = result[i - 1];
+                } else {
+                    result[i] = (${pt.primitive})Math.max(result[i - 1],  v);
```
Comment (kosak) [inline engine/function/src/templates/Numeric.ftl:1624]: Can we remove the extra space before v at the end

File: engine/function/src/templates/Numeric.ftl (hunk 1 above)
Comment (chipkent) [reply engine/function/src/templates/Numeric.ftl:1587]:
```suggestion
                result[i] = (${pt.primitive})Math.max(result[i - 1], values[i]);
```

File: engine/function/src/templates/Numeric.ftl (hunk 2 above)
Comment (chipkent) [reply engine/function/src/templates/Numeric.ftl:1624]:
```suggestion
                    result[i] = (${pt.primitive})Math.max(result[i - 1], v);
```

**Review by kosak — APPROVED** [review]

**Pull request merged**